### PR TITLE
feat(views): Added views to register and unregister a push device

### PR DESCRIPTION
**Pull request by glemmaPaul**
feat(models): Made a method to unregister a push device

feat(urls): Added urls that register the unregister and register endpoints

feat(tests/views): Added tests for the views

feat(tests/models): Changed TestUser to me a BaseAbstractUser

### CODE REVIEW

File: push_notifications/models.py [hunk 1]
```diff
@@ -61,6 +64,11 @@ def register_push_device(cls, user, token, notice_types=None):
         return device
 
     @classmethod
+    def unregister_push_device(cls, user, token):
```
Comment (vpatel85) [inline push_notifications/models.py]: can this be a static method?

File: push_notifications/models.py [hunk 2]
```diff
@@ -61,6 +64,11 @@ def register_push_device(cls, user, token, notice_types=None):
...
+        deleted = user.push_devices.filter(token=token).delete()
+        return deleted > 0
+
+    @classmethod
     def change_permissions(cls, notice_types, device, send=True):
```
Comment (vpatel85) [inline push_notifications/models.py]: same as above.

File: push_notifications/views.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            PushDevice.unregister_push_device(request.user,
+                                              form.cleaned_data['token'])
+            return _send_response({'unregistered': True}, status_code=200)
+
+        # Return back errors
+        data = {
+            'unregistered': False
+        }
+        data.update(form.errors)
+        return _send_response(data, status_code=400)
```
Comment (vpatel85) [inline push_notifications/views.py]: can you return back the errors directly?

Comment (glemmaPaul) [inline push_notifications/views.py]: I think it is a good practice to have `unregistered` be always present, was that what you meant?

Comment (vpatel85) [inline push_notifications/views.py]: Kind of my thought was that `unregistered` would be implied if the call returned a `400` but i like that its explicit.

**Comment by vpatel85** [comment]
:+1: :ship:

**Pull request merged**